build-system/pr-check/build-targets.js

```diff
@@ -247,7 +247,7 @@ function determineBuildTargets(fileName = 'build-targets.js') {
   }
   // Test all targets except VALIDATOR_JAVA on Travis during package upgrades.
   if (isTravisBuild() && buildTargets.has('PACKAGE_UPGRADE')) {
-    const allTargets = Object.keys(targetMatchers);
+    const allTargets = Object.keys(targetMatchers).concat(['RUNTIME']);
```

Comment:
What if RUNTIME was added to targetMatchers in addition to the special cases?

Reply:
`targetMatchers` is the dictionary that maps build targets to matcher functions. For example:

https://github.com/ampproject/amphtml/blob/db85b7102276ff51fd34710a1e7c94da44ece4dd/build-system/pr-check/build-targets.js#L147-L149

We can't add `RUNTIME` to it because it has no equivalent matcher function. It's merely the target to which all files that don't match a different target fall through.

https://github.com/ampproject/amphtml/blob/db85b7102276ff51fd34710a1e7c94da44ece4dd/build-system/pr-check/build-targets.js#L235-L237

I agree this logic seems kinda weird, but I don't have a better idea that doesn't involve a big refactor. Unless you can think of one?

Reply:
Why not match RUNTIME with files that start with src/? It doesn't need to cover all matches since we default to it when none are found.

That way we don't have to keep treating it as a special case in other places.

Reply:
This is an interesting idea. However, we can't have `RUNTIME` simply match everything in `src/` (and I guess `extensions/`) because it will pick up things like tests and OWNERS files, and break the optimizations for reducing the amount of work we do on Travis.

Hard-coding the globs that cover just the runtime code (minus tests, etc.) is brittle because source code directories and files keep getting added.

At that point, we might as well continue to use the _"default to `RUNTIME` if nothing else matches"_ logic. WDYT?

Reply:
I found a non-brittle way of doing this (by using `unitTestPaths`, `integrationTestPaths`, `e2eTestPaths`, and `isOwnersFile()`. Lemme try it before giving up.

Reply:
src/ doesn't include tests though? And just to clarify, by src/ I mean https://github.com/ampproject/amphtml/tree/master/src

To exclude OWNERS files, we can use the pattern already in targetMatchers. And again we don't need to catch everything, so I wouldn't include extensions/.

Reply:
👏 👏 👏 

Did exactly what you suggested.